Lib/unittest/test/testmock/testasync.py

```diff
@@ -52,6 +54,17 @@ def a(self):
 normal_foo_name = f'{__name__}.NormalClass'
 
 
+@contextmanager
+def assertNeverAwaited(test):
+    with test.assertWarns(RuntimeWarning) as cm:
```

Comment:
Personally, using `assertWarnsRegex` looks better since it will give the warning raised in case it didn't match. But I will leave it to you since it's a matter of style. I didn't know the context manager only asserts the first warning so TIL when I use it in future. Other changes look good to me :+1: 

```python
@contextmanager
def assertNeverAwaited(test):
    msg = "was never awaited$"
    with test.assertWarnsRegex(RuntimeWarning, msg) as cm:
        yield
        # In non-CPython implementations of Python, this is needed because timely
        # deallocation is not guaranteed by the garbage collector.
        gc.collect()
```

Added a space after awaited to trigger the below errors : 

```
AssertionError: "was never awaited $" does not match "coroutine 'AsyncMockMixin._execute_mock_call' was never awaited"
```


```
    test.assertTrue(str(cm.warning).endswith('was never awaited '),
AssertionError: False is not true : coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
```

Reply:
Didn't know about assertWarnsRegex, that looks much better!

For the context manager thing, if you think of the code flow of context managers, there's no way for a context manager catching an exception (which warnings are) to return control flow to the point where the exception was raised, so they can only ever deal with one exception in a context...

...of course, this is also why tests should ideally only test one thing per test case ;-)